### PR TITLE
feat(bundling): lower bundle size by discarding comments 

### PR DESCRIPTION
**Pull request by montezume**
#### Summary
While working on #170, I found a quick way to improve our bundle.
We are using [postcss-custom-properties](https://github.com/postcss/postcss-custom-properties) to import css variables from other files. This is a common approach in most of our `mod.css`

```
@import '../../../../materials/colors/base-colors.mod.css';

h1.my-class {
   color: var(--color-green);
}
```

and the current output is as follows.

```

/* stylelint-disable comment-empty-line-before */

/* THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
  This file is created by 'scripts/generate-base-color.js' script. The real
  colors's values should be updated in '../materials/colors/decisions/base-colors.json';
  Also, This file is transformed to JSON version in order to list the colors
  in the UIKit. Color groups are indicated by a comment preceeding the group
  of colors. */

/* Base Colors */

:root {

}

h1.my-class {
   color: green
}

```

This is not ideal, as we end up with these comments (and empty root elements), everywhere. Leading to wasted bytes.

#### Approach

Use [postcss-discard-comments](https://github.com/ben-eb/postcss-discard-comments) to remove the comments. However, if we add this to the end of the plugin list, we will notice that we have removed these comments, but we still have empty root elements like this

```
:root{
}

:root{
}

:root{
}

:root{
}

:root{
}

:root{
}

.number-input-mod_transition-standard__2b_2_{
  -webkit-transition:all 0.2s ease;
  transition:all 0.2s ease;
  -webkit-transition:all 0.2s ease;
  transition:all 0.2s ease;
}
```

One root element per import in the file. We can solve this by placing our discard comment plugin BEFORE the postcss-custom-properties plugin.

Now the output is

```
.number-input-mod_transition-standard__2b_2_{
  -webkit-transition:all 0.2s ease;
  transition:all 0.2s ease;
  -webkit-transition:all 0.2s ease;
  transition:all 0.2s ease;
}
```

Beautiful, isn't it? Lowers the bundle size (esm) to 710KB from 813Kb.

TL:DR -> I lowered the bundle size by over 100kb by removing some css comments 😆 🤣

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -34,6 +35,11 @@ const postcssPlugins = [
     browsers: browserslist.production,
     autoprefixer: { grid: true },
   }),
+  // we need to place the postcssDiscardComments BEFORE postcssCustomProperties,
+  // otherwise we will end up with a bunch of empty :root elements
+  // wherever there are imported comments
+  // see https://github.com/postcss/postcss-custom-properties/issues/123
```
Comment (montezume) [inline rollup.config.js]: this comment is important and I tried to explain it's relevance in the PR description.

Comment (dferber90) [reply rollup.config.js]: Thanks for the comment! I'd also add a link to this PR to the comment as the PR's description does such a nice job of explaining what's going on.
